src/main/java/tools/jackson/databind/deser/bean/BeanDeserializer.java

```diff
@@ -570,7 +570,12 @@ protected Object _deserializeUsingPropertyBased(final JsonParser p, final Deseri
         for (; t == JsonToken.PROPERTY_NAME; t = p.nextToken()) {
             String propName = p.currentName();
             p.nextToken(); // to point to value
-            final SettableBeanProperty creatorProp = creator.findCreatorProperty(propName);
+
+            SettableBeanProperty creatorProp = creator.findCreatorProperty(propName);
+
+            int ix = _propNameMatcher.matchName(propName);
+            SettableBeanProperty regularProp = (ix >= 0) ? _propsByIndex[ix] : null;
```

Comment:
Ok so my problem here is the double lookups in every case, even if in most cases single lookup (for CreatorProperty) should be necessary.
Now, in theory, `_propNameMatcher` lookup should be enough (matcher should be superset of ALL `SettableBeanProperty` entries, and type of property could be checked separately, as needed.

I am trying concurrent changes to do that, but having some problems. Alternatively this PR could be refactored to limit need for extra lookups I think.